From 59f85a9afb15e1d52029f6e0ae30080f15841370 Mon Sep 17 00:00:00 2001
From: Bryan O'Donoghue <bryan.odonoghue@linaro.org>
Date: Tue, 30 Jul 2019 13:10:04 +0100
Subject: [PATCH 2/3] plat-imx: add i.MX6UL Pico-Pi MBL support

Adds support for the i.MX6UL PicoPi MBL configuration.
A straight-forward new board name using

- DTB overlay
- UART6
- NS entry point 0x87800000 instead of 0x80800000

Signed-off-by: Bryan O'Donoghue <bryan.odonoghue@linaro.org>
---
 MAINTAINERS                    |  6 ++++++
 core/arch/arm/plat-imx/conf.mk | 11 +++++++++++
 2 files changed, 17 insertions(+)

diff --git a/MAINTAINERS b/MAINTAINERS
index 2b6585d..f812516 100644
--- a/MAINTAINERS
+++ b/MAINTAINERS
@@ -148,6 +148,12 @@ R:	Etienne Carriere <etienne.carriere@st.com>
 S:	Maintained
 F:	core/arch/arm/plat-stm32mp1/
 
+Technexion PICO-IMX6UL
+R:	Peng Fan <peng.fan@nxp.com> [@MrVan]
+R:	Bryan O'Donoghue <bryan.odonoghue@linaro.org> [@bryanodonoghue]
+S:	Maintained
+F:	core/arch/arm/plat-imx/conf.mk
+
 Texas Instruments AM43xx, AM57xx, DRA7xx, AM65x, J721E
 R:	Andrew F. Davis <afd@ti.com>
 S:	Maintained
diff --git a/core/arch/arm/plat-imx/conf.mk b/core/arch/arm/plat-imx/conf.mk
index eed5d56..93a3f0b 100644
--- a/core/arch/arm/plat-imx/conf.mk
+++ b/core/arch/arm/plat-imx/conf.mk
@@ -4,6 +4,7 @@ PLATFORM_FLAVOR ?= mx6ulevk
 mx6ul-flavorlist = \
 	mx6ulevk \
 	mx6ulccimx6ulsbcpro \
+	mx6ulpico_mbl \
 
 mx6ull-flavorlist = \
 	mx6ullevk \
@@ -172,6 +173,16 @@ CFG_NS_ENTRY_ADDR ?= 0x80800000
 CFG_UART_BASE ?= UART5_BASE
 endif
 
+ifneq (,$(filter $(PLATFORM_FLAVOR),mx6ulpico_mbl))
+CFG_DDR_SIZE ?= 0x20000000
+CFG_NS_ENTRY_ADDR ?= 0x87800000
+CFG_DT_ADDR ?= 0x83100000
+CFG_EXTERNAL_DTB_OVERLAY = y
+CFG_IMX_WDOG_EXT_RESET = y
+CFG_UART_BASE ?= UART6_BASE
+CFG_MX6UL = y
+endif
+
 ifneq (,$(filter $(PLATFORM_FLAVOR),imx8mqevk))
 CFG_DDR_SIZE ?= 0xc0000000
 CFG_UART_BASE ?= UART1_BASE
-- 
2.7.4

